components/brave_shields/browser/https_everywhere_service.cc

```diff
@@ -12,83 +12,70 @@
 
 #include "base/base_paths.h"
 #include "base/bind.h"
-#include "base/json/json_reader.h"
+#include "base/files/file_util.h"
 #include "base/logging.h"
 #include "base/macros.h"
 #include "base/memory/ptr_util.h"
-#include "base/strings/utf_string_conversions.h"
 #include "base/threading/scoped_blocking_call.h"
-#include "base/values.h"
-#include "third_party/leveldatabase/src/include/leveldb/db.h"
-#include "third_party/re2/src/re2/re2.h"
+#include "brave/browser/component_updater/brave_component_installer.h"
+#include "brave/common/pref_names.h"
+#include "brave/vendor/https-everywhere-lib-cpp/src/wrapper.hpp"
+#include "chrome/browser/browser_process_impl.h"
+#include "chrome/browser/first_run/first_run.h"
+#include "components/component_updater/component_updater_service.h"
+#include "components/prefs/pref_service.h"
```

Comment:
I'll need to add `//chrome/browser` and `//brave/browser` to `components/brave_shields/browser/BUILD.gn` for these, but both cause a dependency cycle.
@bridiver any ideas?

Reply:
this is not allowed. You should not be using `g_browser_process` and other things from `//chrome/browser`, `//brave/browser` or `//brave/common`. 
As an example of how to refactor these deps https://github.com/brave/brave-core/pull/6520/files#diff-468a5d483ee258ec61c8b136e8123e16R98 you should pass the local state into the HTTPSEverywhereService constructor